sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -785,7 +793,9 @@ private <T> Slice<T> sliceQuery(SqlQuerySpec querySpec,
 
             maybeEmitEvent(new AfterLoadEvent<>(jsonNode, returnType, containerName));
             final T entity = mappingCosmosConverter.read(returnType, jsonNode);
-            result.add(entity);
+            if (index >= offset) {
```

Comment:
When can index be greater than offset?
Because the above for loop runs index from 0 to pageSize + offset